kratos/includes/table.h

```diff
@@ -521,23 +521,30 @@ class Table<double, double>
     // Get the value for the given argument using piecewise linear
     result_type GetValue(argument_type const& X) const
     {
-        std::size_t size = mData.size();
-
-        KRATOS_ERROR_IF(size == 0) << "Get value from empty table" << std::endl;
-
-        if(size==1) // constant table. Returning the only value we have.
+        int u = mData.size()-1; //using directly u for optimizing one allocation
```

Comment:
```suggestion
        const int u = mData.size()-1; //using directly u for optimizing one allocation
```

Reply:
no, because later on it might be changed in the binary search